Neos.Fusion/Classes/FusionObjects/DataStructureImplementation.php

```diff
@@ -37,7 +37,17 @@ public function evaluate()
         $result = [];
         foreach ($sortedChildFusionKeys as $key) {
             try {
-                $value = $this->fusionValue($key);
+                if (is_array($this->properties[$key])
+                    && !array_key_exists('__objectType', $this->properties[$key])
+                    && !array_key_exists('__eelExpression', $this->properties[$key])
+                    && !array_key_exists('__value', $this->properties[$key])
+                    && !array_key_exists('__stopInheritanceChain', $this->properties[$key])
```

Comment:
I think that could be replaced by sth like:
```php
if (is_array($this->properties[$key]) && empty(array_intersect_key($keys, $this->properties[$key]))
```
where `$keys` contains the reserved words.

But personally I'm more concerned about that taking place within the `try/catch` block.
What about this:


```php
private const RESERVED_KEYS = ['__objectType' => true, '__eelExpression' => true, '__value' => true, '__stopInheritanceChain' => true];

  // ...

            $path = $this->isUntypedProperty($this->properties[$key]) ? $key . '<Neos.Fusion:DataStructure>' : $key;
            try {
                $value = $this->fusionValue($path);
            } catch (\Exception $e) {
                $value = $this->runtime->handleRenderingException($this->path . '/' . $key, $e);
            }

  // ...

    private function isUntypedProperty($property): bool
    {
        if (!is_array($property)) {
            return false;
        }
        return array_intersect_key(self::RESERVED_KEYS, $property) === [];
    }
```
?

Reply:
I just realized, we could also refer to the reserved keys like this:
```php
return array_intersect_key(array_flip(Fusion\Core\Parser::$reservedParseTreeKeys), $property) === [];
```